recipes/openssh/all/conanfile.py

```diff
@@ -52,7 +52,7 @@ def layout(self):
     def requirements(self):
         self.requires("zlib/[>=1.2.11 <2]")
         if self.options.with_libcrypto == "openssl":
-            self.requires("openssl/[>=1.1 <=3.1]")
+            self.requires("openssl/[>=1.1 <3.2]")
```

Comment:
```suggestion
            self.requires("openssl/[>=1.1 <4]")
```

Is there a reason this is capped was capped to 3.1 and now to 3.2? This should have been `[>= 1.1 <4]` as OpenSSL releases are presumed to be backwards compatible, and this is what we are using in all other recipes that use a version range for OpenSSL: 
https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges

Reply:
I have only verified up to and including openssl 3.1.5!
I think a minor version bump requires verification.
When I get to that I will submit!

Reply:
I think this was always meant to be <4 as per https://github.com/conan-io/conan-center-index/pull/19031#discussion_r1508596311 whenI first raised this problem

Reply:
I just committed the suggestion anyway :-)

Reply:
Yes, I think that would be the best course of action, to avoid having to continually bump this.  OpenSSL is API stable as per their guidelines and documentation - they also tend to not change behaviour.

I'm checking the documentation of OpenSSH and I can't find specific versions of OpenSSL/libcrypto being prescribed - however if that turns out to be the case, we could revisit. Otherwise, if you would like to locally ensure that the recipe doesn't cause "new" versions of OpenSSL to be used as they are released, I would recommend using lockfiles for this.